src/main/java/net/masterthought/cucumber/json/deserializers/EmbeddingDeserializer.java

```diff
@@ -26,7 +28,13 @@ public Embedding deserialize(JsonNode rootNode, Configuration configuration) {
         String data = rootNode.get("data").asText();
         String mimeType = findMimeType(rootNode);
 
-        Embedding embedding = new Embedding(mimeType, data);
+        Pattern base64Pattern = Pattern.compile("^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$");
+        Embedding embedding;
+        if (base64Pattern.matcher(data).matches()) {
+            embedding = new Embedding(mimeType, data);
+        } else {
+            embedding = new Embedding(mimeType, new String(Base64.encodeBase64(data.getBytes(UTF_8)), UTF_8));
```

Comment:
EmbeddingDeserializerTest already has test for this so you can simple copy&paste to cover new case

Reply:
I can't find a EmbeddingDeserializerTest.
You sure the class is UnitTested?

Reply:
Right, you can take a look for other reference https://github.com/damianszczepanik/cucumber-reporting/blob/master/src/test/java/net/masterthought/cucumber/json/deserializers/StatusDeserializerTest.java

Reply:
I did that.
The Problem I'm facing is the creation of the embedding files on the filesystem.
As far as I can tell you are using Mockito for mocking, with that I'm not able to mock the static call to Files.